app/controllers/api/pods_controller.rb

```diff
@@ -137,28 +165,56 @@ class PodsController < APIController
           end
         end
 
-        response = version.push!(@owner, specification.to_pretty_json)
-        if response.success?
-          redirect url(version.resource_path)
-        elsif response.failed_on_our_side?
-          throw_internal_server_error!
-        elsif response.failed_on_their_side?
-          # In case of a 5xx at GitHub’s side, this might not mean the commit
-          # didn’t get created, it can also indicate an error occurred while
-          # rendering the response, hence asking for some patience in case we
-          # still update the PodVersion with a new Commit from the GitHub
-          # post-commit hook.
-          #
-          # TODO: Ask GitHub if they have some form of transaction system in
-          # place that rolls back a commit in case an error occurs during
-          # response rendering.
-          json_error(500, 'An error occurred on GitHub’s side. Please check GitHub’s status at ' \
-                          'https://status.github.com and try again later in case the pod is ' \
-                          'still not published.')
-        elsif response.failed_due_to_timeout?
-          json_error(504, 'Calling the GitHub commit API timed out. Please check GitHub’s ' \
-                          'status at https://status.github.com and try again later.')
+        response = version.push!(@owner, specification.to_pretty_json, 'Add')
+        redirect url(version.resource_path) if verify_github_responses!(response)
+      end
+
+      patch '/:name/deprecated', :requires_owner => true do
+        pod = Pod.find_by_name_and_owner(params[:name], @owner) do
+          json_error(403, 'You are not allowed to deprecate this pod.')
+        end
+        unless pod
+          json_error(404, 'No pod found with the specified name.')
+        end
+
+        deprecated_params = JSON.parse(request.body.read)
+        if !deprecated_params.kind_of?(Hash) || deprecated_params.empty?
+          message = 'Please send the deprecation in the body of your post.'
+          json_error(422, message)
+        end
+
+        if in_favor_of = deprecated_params['in_favor_of']
+          unless Pod.find_by_name(Specification.root_name in_favor_of)
+            json_error(422, 'You cannot deprecate a pod in favor of a pod that does not exist.')
```

Comment:
Btw, since we have this check here. Maybe we should have a check somewhere to prevent people pushing pods with deprecated in favour of something else. What we have in trunk already doesn't follow this rule.


```
GexinSDK/1.2.1/GexinSDK.podspec.json:  "deprecated_in_favor_of": "You can use GetuiSDK spec."
HelloWorld/0.1.1/HelloWorld.podspec.json:  "deprecated_in_favor_of": "This was the test library, so i've set a deprecation flag."
```